app/models/notification.rb

```diff
@@ -2,5 +2,26 @@
 
 class Notification < ApplicationRecord
   belongs_to :user
+  enum notification_type: { default: 0, error: 1 }
   default_scope { order(read: :asc, created_at: :desc) }
+
+  def ==(other)
```

Comment:
What is the benefit of using `==` here? The only time this function is used (that I can find) is in the `last_notification == notification` comparison (`app/models/user.rb:59`). This can be replaced with `last_notification.duplicate_of(notification)`, which is arguably clearer.